backend/src/main/java/com/exadel/carinsurance/service/implementation/JwtService.java

```diff
@@ -0,0 +1,65 @@
+package com.exadel.carinsurance.service.implementation;
+
+import com.exadel.carinsurance.service.IJwtService;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.security.Key;
+import java.util.Date;
+
+@Service
+public class JwtService implements IJwtService {
+  @Value( "${jwt.token.secret}" )
+  private String secretKey;
+
+  @Value( "${jwt.token.expirationMs}" )
+  private int expirationMs;
+
+  private Key getSignInKey() {
+    byte[] keyBytes = Decoders.BASE64.decode( secretKey );
+
+    return Keys.hmacShaKeyFor( keyBytes );
+  }
+
+  private Claims extractAllClaims( String token ) {
+    return Jwts
+        .parserBuilder()
+        .setSigningKey( getSignInKey() )
+        .build()
+        .parseClaimsJws( token )
+        .getBody();
+  }

```

Comment:
will throw `io.jsonwebtoken.ExpiredJwtException` if claim is not valid

Reply:
Done